#  Film Industry Analysis for Business Expansion 

## 🎬 **Introduction**  
As part of a company's expansion into the entertainment industry, they are exploring the creation of a new movie studio. However, with no prior experience in film production, it is critical to make data-driven decisions about the types of movies to develop.  

In this project, I will analyze **movie industry data** to answer key business questions such as:  
- **Which movie genres tend to receive the highest ratings?**  
- **Which genres have the highest return on investment (ROI)?**  
- **Are certain directors more likely to deliver high foreign box office success?**  
- **Which film languages are associated with higher popularity and profitability?**  

The goal is to uncover patterns and insights that will help guide the company's decisions on what types of films to produce for maximum commercial success.


## 【1】**Data Exploration**  

### *Loading Datasets*
It is crucial we thoroughly explore each of our datasets and database tables. The datasets we are working with include:
- `Box Office Mojo`
- `IMDB`
- `Rotten Tomatoes`
- `TheMovieDB`
- `The Numbers`
- `im.db` (SQLite database)

Data exploration helps us understand the structure, quality, and characteristics of the data we are working with.  
By examining distributions, missing values, duplicates, and relationships between variables, we can identify potential issues early and plan our cleaning and transformation strategies more effectively.  
This step ensures that our later analyses are based on a solid understanding of the data, reducing the risk of errors and unexpected results.



In [34]:
# Import necessary libraries
import pandas as pd
import sqlite3

In [35]:
# Establish a connection to the database
conn = sqlite3.connect('Data/im.db')

# Check available tables (optional, good for understanding structure)
pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table';", conn)


,name
0,movie_basics
1,directors
2,known_for
3,movie_akas
4,movie_ratings
5,persons
6,principals
7,writers


In [36]:
# close connection
conn.close()

In [42]:
# Load all datasets into pandas DataFrames
bom_df = pd.read_csv('Data/bom.movie_gross.csv')
rt_rev_df = pd.read_csv('Data/rt_critic_reviews.csv')
rt_df = pd.read_csv('Data/rt_movies.csv')
tmdb_df = pd.read_csv('Data/tmdb.movies.csv')
tn_df = pd.read_csv('Data/tn.movie_budgets.csv')

In [38]:
bom_df.head()

,title,studio,domestic_gross,foreign_gross,year
0,Toy Story 3,BV,415000000.0,652000000,2010
1,Alice in Wonderland (2010),BV,334200000.0,691300000,2010
2,Harry Potter and the Deathly Hallows Part 1,WB,296000000.0,664300000,2010
3,Inception,WB,292600000.0,535700000,2010
4,Shrek Forever After,P/DW,238700000.0,513900000,2010


In [43]:
rt_rev_df.head()

,rotten_tomatoes_link,critic_name,top_critic,publisher_name,review_type,review_score,review_date,review_content
0,m/0814255,Andrew L. Urban,False,Urban Cinefile,Fresh,NaN,2010-02-06,A fantasy adventure that fuses Greek mythology...
1,m/0814255,Louise Keller,False,Urban Cinefile,Fresh,NaN,2010-02-06,"Uma Thurman as Medusa, the gorgon with a coiff..."
2,m/0814255,NaN,False,FILMINK (Australia),Fresh,NaN,2010-02-09,With a top-notch cast and dazzling special eff...
3,m/0814255,Ben McEachen,False,Sunday Mail (Australia),Fresh,3.5/5,2010-02-09,Whether audiences will get behind The Lightnin...
4,m/0814255,Ethan Alter,True,Hollywood Reporter,Rotten,NaN,2010-02-10,What's really lacking in The Lightning Thief i...


In [39]:
rt_df.head()

,rotten_tomatoes_link,movie_title,movie_info,critics_consensus,content_rating,genres,directors,authors,actors,original_release_date,...,production_company,tomatometer_status,tomatometer_rating,tomatometer_count,audience_status,audience_rating,audience_count,tomatometer_top_critics_count,tomatometer_fresh_critics_count,tomatometer_rotten_critics_count
0,m/0814255,Percy Jackson & the Olympians: The Lightning T...,"Always trouble-prone, the life of teenager Per...",Though it may seem like just another Harry Pot...,PG,"Action & Adventure, Comedy, Drama, Science Fic...",Chris Columbus,"Craig Titley, Chris Columbus, Rick Riordan","Logan Lerman, Brandon T. Jackson, Alexandra Da...",2010-02-12,...,20th Century Fox,Rotten,49.0,149.0,Spilled,53.0,254421.0,43,73,76
1,m/0878835,Please Give,Kate (Catherine Keener) and her husband Alex (...,Nicole Holofcener's newest might seem slight i...,R,Comedy,Nicole Holofcener,Nicole Holofcener,"Catherine Keener, Amanda Peet, Oliver Platt, R...",2010-04-30,...,Sony Pictures Classics,Certified-Fresh,87.0,142.0,Upright,64.0,11574.0,44,123,19
2,m/10,10,"A successful, middle-aged Hollywood songwriter...",Blake Edwards' bawdy comedy may not score a pe...,R,"Comedy, Romance",Blake Edwards,Blake Edwards,"Dudley Moore, Bo Derek, Julie Andrews, Robert ...",1979-10-05,...,Waner Bros.,Fresh,67.0,24.0,Spilled,53.0,14684.0,2,16,8
3,m/1000013-12_angry_men,12 Angry Men (Twelve Angry Men),Following the closing arguments in a murder tr...,Sidney Lumet's feature debut is a superbly wri...,NR,"Classics, Drama",Sidney Lumet,Reginald Rose,"Martin Balsam, John Fiedler, Lee J. Cobb, E.G....",1957-04-13,...,Criterion Collection,Certified-Fresh,100.0,54.0,Upright,97.0,105386.0,6,54,0
4,m/1000079-20000_leagues_under_the_sea,"20,000 Leagues Under The Sea","In 1866, Professor Pierre M. Aronnax (Paul Luk...","One of Disney's finest live-action adventures,...",G,"Action & Adventure, Drama, Kids & Family",Richard Fleischer,Earl Felton,"James Mason, Kirk Douglas, Paul Lukas, Peter L...",1954-01-01,...,Disney,Fresh,89.0,27.0,Upright,74.0,68918.0,5,24,3


In [40]:
tmdb_df.head()

,Unnamed: 0,genre_ids,id,original_language,original_title,popularity,release_date,title,vote_average,vote_count
0,0,"[12, 14, 10751]",12444,en,Harry Potter and the Deathly Hallows: Part 1,33.533,2010-11-19,Harry Potter and the Deathly Hallows: Part 1,7.7,10788
1,1,"[14, 12, 16, 10751]",10191,en,How to Train Your Dragon,28.734,2010-03-26,How to Train Your Dragon,7.7,7610
2,2,"[12, 28, 878]",10138,en,Iron Man 2,28.515,2010-05-07,Iron Man 2,6.8,12368
3,3,"[16, 35, 10751]",862,en,Toy Story,28.005,1995-11-22,Toy Story,7.9,10174
4,4,"[28, 878, 12]",27205,en,Inception,27.920,2010-07-16,Inception,8.3,22186


In [41]:
tn_df.head()

,id,release_date,movie,production_budget,domestic_gross,worldwide_gross
0,1,"Dec 18, 2009",Avatar,"$425,000,000","$760,507,625","$2,776,345,279"
1,2,"May 20, 2011",Pirates of the Caribbean: On Stranger Tides,"$410,600,000","$241,063,875","$1,045,663,875"
2,3,"Jun 7, 2019",Dark Phoenix,"$350,000,000","$42,762,350","$149,762,350"
3,4,"May 1, 2015",Avengers: Age of Ultron,"$330,600,000","$459,005,868","$1,403,013,963"
4,5,"Dec 15, 2017",Star Wars Ep. VIII: The Last Jedi,"$317,000,000","$620,181,382","$1,316,721,747"


After successfully loading all the datasets using pandas, we are now ready to dive deeper into data exploration. Given that we are working with six distinct datasets and a database, manually using `.info()`, `.describe()`, `.isna()`, and writing separate code blocks to inspect every dataframe would be highly inefficient, tedious and extremely confusing. Moreover, for the `im.db` database, exploring each table one by one through traditional code would have required an overwhelming amount of setup, making the process unnecessarily complex.

To overcome this, we identified tools that allowed us to explore our data in a far more efficient and highly visual manner.  


### *Data Exploration with Data Wrangler and SQLite Viewer*

For the initial exploration of our datasets, we utilized two powerful Visual Studio Code extensions: 

`Data Wrangler` made early-stage data exploration much faster and more intuitive by providing:
- Visual summaries of column distributions and data types.
- Instant detection of missing values, outliers, and duplicates.
- Quick sorting, filtering, and grouping operations without needing to write code.
- Profiling of datasets with automated suggestions for cleaning and transformation.

`SQLite Viewer` allowed us to efficiently browse the database structure and contents without having to write extensive SQL queries.  
With it, we could:
- Instantly preview tables, columns, and sample data.
- Navigate between tables and their relationships quickly.
- Understand data volumes, missing values, and schema designs in a highly visual and interactive way.

Using `Data Wrangler` and `SQLite Viewer` streamlined our workflow significantly.  
Instead of manually traversing large datasets and databases line-by-line in code, we were able to focus our time on identifying meaningful issues, relationships, and insights

<div style="white-space: nowrap; overflow-x: auto;">
    <img src="images/bom_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/rt_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/rt_rev_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/tmdb_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/tn_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/im.db.png" style="display: inline-block; height: 500px; margin-right: 10px;">
</div>

After thorough exploration and exhaustive discussion of the datasets, we carefully evaluated the available columns. Based on importance to our project goals and quality, we decided on a final set of columns that would be most effective for a meaningful analysis:

- From `rotten_tomatoes`:
  - `movie_title`, `genres`, `tomatometer_rating`, `audience_rating`  
    → These columns help us evaluate which movie genres tend to receive the highest critical and audience ratings.

- From `tmdb`:
  - `language`, `popularity`  
    → These attributes allow us to investigate how a film's language impacts its popularity and, by extension, potential profitability.

- From `the_numbers`:
  - `movie`, `production_budget`, `worldwide_gross`  
    → These financial figures are crucial for calculating Return on Investment (ROI) and understanding broader profitability patterns across genres.

- From `im.db`:
  - Extracting and joining data across the `directors`, `persons`, and `movie_akas` tables  
    → This allows us to match each director with the correct movie title, enabling the analysis of which directors tend to deliver high foreign box office success.

Even though these are the specific columns we are interested in, it is still necessary to properly prepare and relate the data across different datasets to build a strong foundation for answering the business questions effectively.

## 【1】**Data Exploration & Cleaning**  

### **rt.ratings and genre data prep**

#### **Rotten Tomatoes Movies Dataset Description**

The **Rotten Tomatoes Movies** dataset provides detailed information about a wide range of movies, collected from the Rotten Tomatoes website.  
It contains metadata that can be used to analyze movie characteristics, trends, audience perceptions, and more.

---

#### **Key features**
- **movie_title** The title of the movie as listed on Rotten Tomatoes.
- **genres** A comma-separated list of genres assigned to the movie (e.g., Drama, Comedy, Action).
- **audience_rating** The average rating given by the audience, typically scored from 0 to 100.
- **tomatometer_rating** The percentage of positive critics rating


---

#### **Key Points**
- Some movies belong to **multiple genres** (e.g., "Action, Comedy, Drama").
- Not all entries have complete information; there might be **missing values** for some fields.
- The audience rating reflects the **general public's** reception rather than professional critics' reviews.
- Some text fields (like movie titles and genres) may have **inconsistent formatting** (e.g., upper vs lower case).

---




### **step 1: reading the dataset**

In [33]:
#importing the necessary libraries 
import numpy as np
import pandas as pd

We import essential Python libraries:
- **NumPy** for numerical operations.
- **Pandas** for data manipulation and analysis.


In [34]:
# loading the dataset
rt_movies=pd.read_csv(r"C:\Users\HomePC\Desktop\phase_2-project\dsc-phase-1-project-template\rotten_tomatoes_movies.csv")
rt_movies.sample(5)

,rotten_tomatoes_link,movie_title,movie_info,critics_consensus,content_rating,genres,directors,authors,actors,original_release_date,...,production_company,tomatometer_status,tomatometer_rating,tomatometer_count,audience_status,audience_rating,audience_count,tomatometer_top_critics_count,tomatometer_fresh_critics_count,tomatometer_rotten_critics_count
4277,m/cant_think_straight,I Can't Think Straight,"As her wedding day approaches, a feisty bride ...",NaN,R,"Drama, Romance",Shamim Sarif,"Shamim Sarif, Kelly Moss","Lisa Ray, Sheetal Sheth, Antonia Frering, Dali...",2008-06-01,...,Regent Releasing/here! Films,Rotten,16.0,32.0,Upright,62.0,36879.0,12,5,27
9090,m/koch_2012,Koch,The life of former New York City Mayor Ed Koch.,NaN,NR,"Documentary, Special Interest",Neil Barsky,NaN,"Ed Koch, Michael Bloomberg, Calvin O. Butts II...",2013-02-01,...,Zeitgeist Films,Fresh,90.0,39.0,Upright,76.0,395.0,17,35,4
8358,m/inferno_2016,Inferno,Famous symbologist Robert Langdon (Tom Hanks) ...,"Senselessly frantic and altogether shallow, In...",PG-13,"Action & Adventure, Drama, Mystery & Suspense",Ron Howard,"Ron Howard, David Koepp","Tom Hanks, Felicity Jones, Irrfan Khan, Omar S...",2016-10-28,...,Sony Pictures,Rotten,23.0,250.0,Spilled,36.0,28773.0,50,58,192
6037,m/entangled_2019,Entangled,A young French woman living in New York City b...,NaN,NR,Drama,Milena Lurie,Milena Lurie,"Jonathan Cake, Lucy Walters, Theodora Miranne,...",2019-10-11,...,Samuel Goldwyn Films,Rotten,20.0,5.0,NaN,NaN,NaN,0,1,4
17200,m/what_the__do_we_know,What the Bleep Do We Know!?,"Amanda, a divorced photographer, has a fantast...","Educational, but as entertainment, it'll have ...",NR,"Documentary, Drama, Science Fiction & Fantasy,...","Marc Vicente, William Arntz, Betsy Chasse, Mar...","Matt Hoffman, Matthew Hoffman, Betsy Chasse, W...","Marlee Matlin, Barry Newman, Elaine Hendrix, A...",2004-04-23,...,Lord of the Wind,Rotten,34.0,77.0,Upright,69.0,22052.0,25,26,51


We load the movie data from a CSV file into a pandas DataFrame called `rt_movies`.  
A sample of 5 random rows is displayed to understand the structure of the dataset.


#### **step 2 extracting the relevant columns and exploding the genre column**


We select only three columns: `movie_title`, `genres`,`tomatometer_rating` and `audience_rating` because these are the columns we will need in our analysis.
- The `genres` column contains multiple genres separated by commas.  
- We **split** this column into lists so that each genre can later be separated into individual rows.


In [35]:
#extracting the relevant columns
df_rt=rt_movies[["movie_title","genres","audience_rating","tomatometer_rating"]]
# Convert the comma-separated string into a list
df_rt['genres'] = df_rt['genres'].str.split(', ')

# exploding the genres
rt_movies_exploded = df_rt.explode('genres').reset_index(drop=True)
rt_movies_exploded.head()



C:\Users\HomePC\AppData\Local\Temp\ipykernel_12892\2531300135.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_rt['genres'] = df_rt['genres'].str.split(', ')


,movie_title,genres,audience_rating,tomatometer_rating
0,Percy Jackson & the Olympians: The Lightning T...,Action & Adventure,53.0,49.0
1,Percy Jackson & the Olympians: The Lightning T...,Comedy,53.0,49.0
2,Percy Jackson & the Olympians: The Lightning T...,Drama,53.0,49.0
3,Percy Jackson & the Olympians: The Lightning T...,Science Fiction & Fantasy,53.0,49.0
4,Please Give,Comedy,64.0,87.0


We **explode** the `genres` column:
- Each genre becomes a separate row.
- This transforms one-to-many relationships into one-to-one, making it easier to analyze by individual genre.
- After exploding, we reset the index to clean up the DataFrame.


In [36]:
#comparing the number of rows for main df and the exploded df 
print(f"main:{rt_movies.shape[0]} rows")
print(f"exploded:{rt_movies_exploded.shape[0]} rows")


main:17712 rows
exploded:39388 rows


We compare the number of rows in:
- The original DataFrame
- The exploded DataFrame

An increase in rows confirms that multiple genres per movie were successfully separated into multiple rows.


In [37]:
#seeing if the number of unique movie_title values matches with the main
print(f"exploded: {len(rt_movies_exploded['movie_title'].unique())} unique values")
print(f"main df: {len(rt_movies['movie_title'].unique())} unique values")


exploded: 17106 unique values
main df: 17106 unique values


We verify that the number of unique `movie_title` entries remains consistent between the original and exploded datasets.
This ensures that exploding the genres did not accidentally duplicate or lose any movies.



### **step 3: summarizing the dataset**

Summarizing a DataFrame is essential to **quickly understand the dataset** and spot issues early.  
It helps in:

- **Understanding structure**: Number of rows, columns, and their types.
- **Detecting missing values**: Finding gaps that could affect analysis.
- **Checking data types**: Ensuring columns are correctly formatted for operations.
- **Assessing memory usage**: Optimizing performance for large datasets.
- **Identifying potential problems**: Catching inconsistencies like wrong formats or placeholders.
- **Guiding cleaning steps**: Deciding what transformations are needed.
- **Saving time**: Providing a quick overview instead of manual checking.

---


In [38]:
# summary of the explode dataset
rt_movies_exploded.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39388 entries, 0 to 39387
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   movie_title         39388 non-null  object 
 1   genres              39369 non-null  object 
 2   audience_rating     38899 non-null  float64
 3   tomatometer_rating  39296 non-null  float64
dtypes: float64(2), object(2)
memory usage: 1.2+ MB


the dataframe has **39388 rows and 3 columns**. `genres`,`audience_rating`, and `tomatometer_rating` **have missing values** but the missing values  are not significant. `audience rating` and `tomatometer rating` columns are **numerical** while `movie title` and the `genres` colums are **categorical**.
all the column values have the same **data types** as the ones we expect them to have

### **step 4: description of the dataframe and checking for outliers**

#### **dataframe description**

Describing a DataFrame gives a **statistical summary** of numerical columns, providing **deeper insights** into the data distribution and behaviour.

It helps in:
- Understanding Data Distribution
- Detecting Outliers
- Checking Data Variability
- Validating Data Accuracy
- Supporting Feature Engineering
- Assessing Skewness




In [39]:
##describing the numerical column
rt_movies_exploded.describe()


,audience_rating,tomatometer_rating
count,38899.000000,39296.000000
mean,61.542945,62.107619
std,20.373430,28.047419
min,0.000000,0.000000
25%,46.000000,40.000000
50%,64.000000,67.000000
75%,78.000000,86.000000
max,100.000000,100.000000


This table compares the distribution of **Audience Ratings** and **Tomatometer Ratings** from a sample of movie reviews. The statistics highlight differences in central tendency, spread, and skew between general audience scores and critic-aggregated ratings.

- **Sample Size**:  
  - Audience Ratings: **38,899** reviews  
  - Tomatometer Ratings: **39,296** reviews  
- **Average Ratings**:  
  - Audiences tend to rate slightly lower (**61.54**) compared to critics (**62.11**).  
- **Variability**:  
  - Critics' ratings exhibit higher variability (**std = 28.05**) than audience ratings (**std = 20.37**).  
- **Distributions**:  
  - Both metrics range from **0–100**, but critics' ratings are more polarized:  
    - 25% of critic scores are ≤ **40**, while 75% of audience scores are ≤ **78**.  
    - Median critic ratings (**67**) are higher than audience medians (**64**).  
  

#### **checking for outliers**

Using the Interquartile Range (IQR) method, we check for outliers:
- **IQR = Q3 - Q1**
- Any data point below **Q1 - 1.5 * IQR** or above **Q3 + 1.5 * IQR** is considered an outlier.

We apply this logic to:
- `audience_rating`
- `tomatometer_rating`

Detecting and handling outliers is crucial for reliable statistical analysis.


In [40]:
#checking for outliers
def check_outliers(q1,q3,min,max):
    iqr=q3-q1
    upper_bound=q3+(1.5*iqr)
    lower_bound=q1-(1.5*iqr)
    if (min>lower_bound) & (max<upper_bound):
     return("There are no outliers")
    else:
     return("There are outliers")
# checking in the audience_rating column
print(f"{check_outliers(q1=46,q3=78,min=0,max=100)} in the audience rating column")
# checking in the tomatometer_rating column
print(f"{check_outliers(q1=40,q3=86,min=0,max=100)} in the tomatometer rating column")


There are no outliers in the audience rating column
There are no outliers in the tomatometer rating column


### **step 5 : checking and handling missing values**

Understanding missing data is important to Decide whether to remove or impute missing values which will maintain data integrity for further analysis.


#### **checking for missing values**

In [41]:
# checking for missing values in terms of percentage
(rt_movies_exploded.isna().sum()/len(rt_movies_exploded)*100)

movie_title           0.000000
genres                0.048238
audience_rating       1.241495
tomatometer_rating    0.233574
dtype: float64

`genres`,`audience_rating` and `tomatometer_rating` have missing values in the respective columns, but the missing values are insignificant compared to the total number of entries.
We drop all the rows with missing values since dropping them won't impact our analyses.

#### **handling missing values**

In [42]:
# dropping the rows which have empty values under the columns
rt_movies_exploded_cleaned=rt_movies_exploded.dropna(axis=0,subset="genres")
rt_movies_exploded_cleaned.dropna(axis=0,subset="audience_rating",inplace=True)
rt_movies_exploded_cleaned.dropna(axis=0,subset="tomatometer_rating",inplace=True)
# checking if there is still any missing values left
rt_movies_exploded_cleaned.isna().sum()

C:\Users\HomePC\AppData\Local\Temp\ipykernel_12892\1016391485.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned.dropna(axis=0,subset="audience_rating",inplace=True)
C:\Users\HomePC\AppData\Local\Temp\ipykernel_12892\1016391485.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned.dropna(axis=0,subset="tomatometer_rating",inplace=True)


movie_title           0
genres                0
audience_rating       0
tomatometer_rating    0
dtype: int64

After dropping the missing values, the dataframe now does not have any missing values

In [43]:
## checking for the number of rows that reduced after handling missing values
reduced=rt_movies_exploded_cleaned.shape[0]-rt_movies_exploded.shape[0]
percent_reduced=round((reduced/len(rt_movies_exploded))*100)
print(f"percentage reduced={percent_reduced} percent")



percentage reduced=-1 percent


after cleaning the dataframe the percentage of the rows that reduced from the original dataframe is 1 percent hence it will not really affect our analyses 

#### **step 6: checking for placeholders**

checking and handling placeholders improves the integrity of the data

In [44]:
# checking for placeholders in the genres column
rt_movies_exploded_cleaned["genres"].value_counts()

genres
Drama                        9432
Comedy                       5593
Action & Adventure           3526
Mystery & Suspense           3502
Art House & International    2476
Horror                       2023
Documentary                  1960
Romance                      1897
Science Fiction & Fantasy    1845
Classics                     1647
Special Interest             1211
Kids & Family                1109
Musical & Performing Arts     989
Animation                     624
Western                       301
Television                    265
Sports & Fitness              175
Cult Movies                   104
Faith & Spirituality           79
Gay & Lesbian                  76
Anime & Manga                  19
Name: count, dtype: int64

there was no placeholder in the genres column.
since the minimum value in the audience_rating and tomatometer_rating is 0 in each column and the maximum value is 100, we can conclude that they have no placeholders.

### **step 7: checking for duplicates**

Duplicate rows can bias results hence they should be removed to maintain data quality


In [45]:
#checking for duplicates
print(f"{rt_movies_exploded_cleaned.duplicated().sum()} duplicates")


0 duplicates


the cleaned dataframe has no duplicates

#### **step 8: standardizing text columns**

We standardize text data by converting `movie_title` and `genres` columns to lowercase.
Standardization helps avoid inconsistencies during grouping, searching, or filtering.
(e.g., "Drama" and "drama" would otherwise be treated differently)


In [46]:
rt_movies_exploded_cleaned["genres"]=rt_movies_exploded_cleaned["genres"].str.lower()
rt_movies_exploded_cleaned["movie_title"]=rt_movies_exploded_cleaned["movie_title"].str.lower()
rt_movies_exploded_cleaned.sample(5)


C:\Users\HomePC\AppData\Local\Temp\ipykernel_12892\2608153190.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned["genres"]=rt_movies_exploded_cleaned["genres"].str.lower()
C:\Users\HomePC\AppData\Local\Temp\ipykernel_12892\2608153190.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rt_movies_exploded_cleaned["movie_title"]=rt_movies_exploded_cleaned["movie_title"].str.lower()


,movie_title,genres,audience_rating,tomatometer_rating
31568,sunset,mystery & suspense,30.0,23.0
22232,mad hot ballroom,documentary,83.0,84.0
38082,the wedding date,romance,67.0,11.0
32278,that sugar film,documentary,75.0,68.0
19479,the jewel of the nile,action & adventure,37.0,48.0


### **step 9: summary after cleaning**

We run `.info()` once again after all cleaning steps to:
- Confirm the dataset is fully clean
- Verify column types
- Validate the number of entries


In [47]:
## doing a summary
rt_movies_exploded_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Index: 38853 entries, 0 to 39387
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   movie_title         38853 non-null  object 
 1   genres              38853 non-null  object 
 2   audience_rating     38853 non-null  float64
 3   tomatometer_rating  38853 non-null  float64
dtypes: float64(2), object(2)
memory usage: 1.5+ MB


The cleaned dataframe:
- has no missing values
- have the correct data types

this step allows the data to be reused without preprocessing

### genres and ROI data prep
#### Vincent can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

### language and popularity data prep
#### Chris can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

### Preparing Directors and Foreign Gross Data

We will prepare and align two datasets to explore the relationship between directors and the worldwide gross earnings of their movies. From our earlier data exploration, we have confirmed that there are no missing values in the relevant `im.db` tables and in the Numbers movie budgets dataset, which will allow us to proceed without needing additional imputation steps. Since these datasets originate from different sources and structures, we will need to carefully filter and merge them to ensure accuracy.  

In [26]:
import sqlite3
import pandas as pd

#### *Extracting Director and Movie Title Data*

We will start by extracting a clean dataset containing each director's name and their corresponding movie title from the `im.db` database.  
To ensure we retrieve only unique director–movie pairs, we are using a `DISTINCT` query.  
During earlier exploration, we noticed several duplicate entries caused by the way data was entered into the database, so applying `DISTINCT` helps eliminate these redundancies by focusing only on original movie titles.


In [27]:
# Connect to the database
conn = sqlite3.connect('Data/im.db')

# Updated query with DISTINCT
query = '''
SELECT DISTINCT
    p.primary_name AS director_name,
    ma.title AS movie_title
FROM directors d
JOIN persons p ON d.person_id = p.person_id
JOIN movie_akas ma ON d.movie_id = ma.movie_id
WHERE ma.is_original_title = 1
'''

# Execute the query and load into a DataFrame
directors_df = pd.read_sql_query(query, conn)

# Close connection
conn.close()
directors_df.head()


,director_name,movie_title
0,Colin Trevorrow,Jurassic World
1,Andrew Stanton,John Carter
2,Marc Jampolsky,Versailles Rediscovered - The Sun King's Vanis...
3,Sam Zubrycki,Miguelito - Canto a Borinquen
4,Henning Beckhoff,Thing I Don't Get


In [28]:
tn_df = pd.read_csv("Data/tn.movie_budgets.csv")
tn_df.columns.tolist()

['id',
 'release_date',
 'movie',
 'production_budget',
 'domestic_gross',
 'worldwide_gross']

#### *Merging Directors with Worldwide Gross Data*

We need to combine the director information with the worldwide gross earnings.  
This step ensures that each movie in our dataset is matched with both its director and its revenue performance, setting the foundation for meaningful analysis.


In [32]:
# Rename for easy merge
directors_df = directors_df.rename(columns={'movie_title': 'movie'})

# Merge the two DataFrames
merged_df = pd.merge(tn_df[['movie', 'worldwide_gross']], directors_df, on='movie', how='inner')

# Reorder columns
directors_fgross_df = merged_df[['movie', 'director_name', 'worldwide_gross']]

# Sort alphabetically by director name
directors_fgross_df = directors_fgross_df.sort_values(by='director_name').reset_index(drop=True)
directors_fgross_df


,movie,director_name,worldwide_gross
0,Akira,A.R. Murugadoss,"$19,585"
1,Bully,Aaron Alon,"$1,381,824"
2,Restless,Aaron Boltz,"$2,772,511"
3,Circle,Aaron Hann,"$10,024"
4,Vampires Suck,Aaron Seltzer,"$81,424,988"
...,...,...,...
2381,Witness,Zehao Gao,"$65,532,576"
2382,Along the Roadside,Zoran Lisinac,"$3,234"
2383,Prophecy,Zuri Rinpoche,"$22,673,340"
2384,9,Éric Tessier,"$48,559,999"


#### *Aggregating Worldwide Gross Earnings by Director*

We need to aggregate the data to account for the fact that a single director may have multiple movies. Our goal is to sum up the worldwide gross earnings for each director across all their movies, creating a table that shows the total earnings per director. This aggregation will give us a clearer picture of each director's overall financial impact.


In [ ]:
# Remove commas, dollar sign and convert to float for calculation
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r',', '', regex=True)
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r'\$', '', regex=True).astype(float)

# Group by director_name and sum worldwide_gross
agg_directors_df = directors_fgross_df.groupby('director_name', as_index=False)['worldwide_gross'].sum()

# Sort by worldwide_gross in descending order
agg_directors_df = agg_directors_df.sort_values(by='worldwide_gross', ascending=False).reset_index(drop=True)

# Format numbers with commas 
agg_directors_df['worldwide_gross'] = agg_directors_df['worldwide_gross'].apply(lambda x: f"{int(x):,}")
agg_directors_df


,director_name,worldwide_gross
0,Joe Russo,"3,902,605,502"
1,Anthony Russo,"3,902,605,502"
2,Pierre Coffin,"3,713,745,331"
3,Christopher Nolan,"3,086,180,484"
4,Joss Whedon,"2,992,084,614"
...,...,...
1651,Conor Allyn,0
1652,Curt Hahn,0
1653,Corbin Bernsen,0
1654,Varante Soudjian,0


## 【2】**Data Analysis**  

#### *1. Data Analysis*  

#### *2. Hypothesis Testing*  

## 【3】**Business Recommendations**  

## 【4】**Limitations of Our Analysis**  